code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1050 @@
+		name   string
+		client *http.Client
+		args   args
+	}{
+		{
+			name:   "OK",
+			client: server.Client(),
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			httpClient = tt.client
+
+			SendMultiPostRequest(tt.args.urls, tt.args.data, tt.args.ID, tt.args.pkey)
+		})
+	}
+}
+
+func TestSendTransaction(t *testing.T) {
+	t.Parallel()
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+			},
+		),
+	)
+	defer server.Close()
+
+	type args struct {
+		txn  *Transaction
+		urls []string
+		ID   string
+		pkey string
+	}
+	tests := []struct {
+		name   string
+		args   args
+		client *http.Client
+	}{
+		{
+			name:   "OK",
+			client: server.Client(),
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			httpClient = tt.client
+
+			SendTransaction(tt.args.txn, tt.args.urls, tt.args.ID, tt.args.pkey)
+		})
+	}
+}
+
+func TestMakeGetRequest(t *testing.T) {
+	t.Parallel()
+
+	t.Skip("need to protect global variable httpClient against concurrent access")
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				data := map[string]interface{}{
+					"key": "value",
+				}
+				blob, err := json.Marshal(data)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer server.Close()
+
+	type args struct {
+		remoteUrl string
+		result    interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		client  *http.Client
+		wantErr bool
+	}{
+		{
+			name: "Request_Creating_ERR",
+			args: args{
+				remoteUrl: string(rune(0x7f)),
+			},
+			wantErr: true,
+		},
+		{
+			name:   "Client_ERR",
+			client: server.Client(),
+			args: args{
+				remoteUrl: "/",
+			},
+			wantErr: true,
+		},
+		{
+			name:   "Resp_Status_Not_Ok_ERR",
+			client: errServer.Client(),
+			args: args{
+				remoteUrl: errServer.URL,
+			},
+			wantErr: true,
+		},
+		{
+			name:   "JSON_Decoding_ERR",
+			client: server.Client(),
+			args: args{
+				remoteUrl: server.URL,
+				result:    "}{",
+			},
+			wantErr: true,
+		},
+		{
+			name:   "OK",
+			client: server.Client(),
+			args: args{
+				remoteUrl: server.URL,
+				result:    &map[string]interface{}{},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			httpClient = tt.client
+
+			if err := MakeGetRequest(tt.args.remoteUrl, tt.args.result); (err != nil) != tt.wantErr {
```

Comment:
"OK" test fails:
`MakeGetRequest` looks difficult to test. It gets an http request with [`NewHTTPRequest`](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/chaincore/httpclientutil/http_client_util.go#L88), which uses `http.NewReqeust` were we want to use `httptest.NewRequest`. I can't see an easy way of mocking NewHTTPRequest.

Reply:
I've removed redefining and parallel running because `OK` test has strange falls with parallel running. Now it works, can you please recheck this on your local machine?